app/models/course/assessment.rb

```diff
@@ -135,10 +135,18 @@ def initialize_duplicate(duplicator, other)
     copy_attributes(other, duplicator.time_shift)
     self.folder = duplicator.duplicate(other.folder)
     self.questions = duplicator.duplicate(other.questions.map(&:actable)).compact.map(&:acting_as)
-    self.assessment_conditions = duplicator.duplicate(other.assessment_conditions)
-    # Like achievement conditions, duplicate the actable object directly and let the acting_as
-    # gem create the Condition object.
-    self.conditions = duplicator.duplicate(other.conditions.map(&:actable)).map(&:acting_as)
+    if duplicator.mode == :course
+      self.assessment_conditions = duplicator.duplicate(other.assessment_conditions)
+      # Like achievement conditions, duplicate the actable object directly and let the acting_as
+      # gem create the Condition object.
+      self.conditions = duplicator.duplicate(other.conditions.map(&:actable)).map(&:acting_as)
+    elsif duplicator.mode == :object
+      target_category = duplicator.options[:target_course].assessment_categories.first
```

Comment:
At some point in the future this list of arguments might be pretty un-maintainable, which was why I preferred to use options (not sure if there are other possible 'modes'). This also requires work on the `Duplicator` too, so I'm not sure if the suggested way is cleaner. 

What do you think?